app/src/main/java/rick_and_morty/data/repository/CharacterRepository.kt

```diff
@@ -0,0 +1,24 @@
+package rick_and_morty.data.repository
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import rick_and_morty.di.RickAndMortyApiRemoteDataSource
+import javax.inject.Inject
+import rick_and_morty.data.model.CharacterResultsDto
+
+
+class CharacterRepository @Inject constructor(
+    private val rickAndMortyApiRemoteDataSource: RickAndMortyApiRemoteDataSource
+){
+    
+   fun getCharacters(page: Int): Flow<List<CharacterResultsDto>> = flow {
+        emit(rickAndMortyApiRemoteDataSource.fetchRickAndMortyData(page).results!!)
```

Comment:
In case there will be no `result` available the app would break here. That's why checking the data model is necessary.